ethcore/src/verification/queue/kind.rs

```diff
@@ -119,14 +119,13 @@ pub mod blocks {
 
 	impl Unverified {
 		/// Create an `Unverified` from raw bytes.
-		pub fn new(bytes: Bytes) -> Self {
-			use views::BlockView;
+		pub fn from_rlp(bytes: Bytes) -> Result<Self, ::rlp::DecoderError> {
 
-			let header = view!(BlockView, &bytes).header();
-			Unverified {
+			let header = ::rlp::Rlp::new(&bytes).val_at(0)?; 
```

Comment:
Do you mind fixing similar issue for `ancient block` (`block_with_receipt`) import?
AFAIR we also have a `BlockView` there.

Reply:
Also `collect_blocks` (`https://github.com/paritytech/parity/blob/master/ethcore/sync/src/block_sync.rs#L472`). It's the only place that calls `import_block_with_receipts` but it creates a header view before doing so.

Reply:
Ok I think what comes from `BlockCollection` is valid (invalid blocks/headers are silently ignored when inserting) so `collect_blocks` is safe, not sure if we should update it.

Reply:
In fact, invalid header/block body RLP is rejected further upstream in [`block_sync`](https://github.com/paritytech/parity/blob/master/ethcore/sync/src/block_sync.rs#L234), which results in the offending peer [being deactivated](https://github.com/paritytech/parity/blob/master/ethcore/sync/src/chain.rs#L779).

So blocks pulled from `BlockCollection` should all be valid RLP so safe to use the views inside `collect_blocks`. 

Seems to me that we could remove plenty of 'unsafe' view usages in this path since we could pass down the pre-decoded headers/block bodies, alongside/instead of the raw bytes. But that would be a separate PR.

Reply:
@tomusdrw see https://github.com/paritytech/parity/pull/8522/commits/22b2c3a7562ea1caa5379eb5d91b0b9d2957299a